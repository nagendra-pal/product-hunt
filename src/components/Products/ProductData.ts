import moment from "moment";
import {ProductType} from "./ProductTypes";

const ProductData: Array<ProductType> = [
    {
        "id" : 1,
        "name": "Wireless headphones",
        "image": "https://res.cloudinary.com/dsrkwxxo7/image/upload/v1589218153/samples/products/Wireless-headphones_vxnlz2.jpg",
        "desc": "Lorem ipsum dolor amet post-ironic four dollar toasts selvage meh Lorem ipsum dolor amet post-ironic four dollar toasts selvage meh Lorem ipsum dolor amet post-ironic four dollar toasts selvage meh",
        "likes": 4,
        "comments": [
            {
                "text": "Lorem ipsum dolor sit amet, consectetur adipisicing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua."
            },
            {
                "text": "Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat." 
            },
            {
                
                "text": "Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur."
                
            },
            {
                
                "text": "Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum."
            }
        ],
        "created": moment.utc("2020-03-25").toDate()
    },
    {
        "id" : 2,
        "name": "Round white watch with white band",
        "image": "https://res.cloudinary.com/dsrkwxxo7/image/upload/v1589218152/samples/products/White-watch_n3ansu.jpg",
        "desc": "Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur.",
        "likes": 10,
        "comments": [
            {
                
                "text": "Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur."
                
            },
            {
                
                "text": "Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum."
            }
        ],
        "created": moment.utc("2020-04-10").toDate()
    },
    {
        "id" : 3,
        "name": "Unpaired maroon plimsoll on top of yellow textile",
        "image": "https://res.cloudinary.com/dsrkwxxo7/image/upload/v1589218152/samples/products/Maroon-plimsoll_i8lhlr.jpg",
        "desc": "Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur.",
        "likes": 5,
        "comments": [
            {
                "text": "Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat." 
            }
        ],
        "created": moment.utc("2020-04-15").toDate()
    },
    {
        "id" : 4,
        "name": "Beige wooden bar stool",
        "image": "https://res.cloudinary.com/dsrkwxxo7/image/upload/v1589218152/samples/products/Wooden-stool_bex90w.jpg",
        "desc": "Lorem ipsum dolor amet post-ironic four dollar toasts selvage meh Lorem ipsum dolor amet post-ironic four dollar toasts selvage meh Lorem ipsum dolor amet post-ironic four dollar toasts selvage meh",
        "likes": 20,
        "comments": [
            {
                "text": "Lorem ipsum dolor sit amet, consectetur adipisicing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua."
            },
            {
                "text": "Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat." 
            },
            {
                
                "text": "Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur."
                
            },
            {
                
                "text": "Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum."
            },
            {
                "text": "Lorem ipsum dolor sit amet, consectetur adipisicing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua."
            },
            {
                "text": "Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat." 
            },
            {
                
                "text": "Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur."
                
            },
            {
                
                "text": "Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum."
            }
        ],
        "created": moment.utc("2020-04-20").toDate()
    },
    {
        "id" : 5,
        "name": "Black sunglasses",
        "image": "https://res.cloudinary.com/dsrkwxxo7/image/upload/v1589218152/samples/products/Black-sunglasses_mffok6.jpg",
        "desc": "Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat.",
        "likes": 7,
        "comments": [
            {
                
                "text": "Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur."
                
            },
            {
                
                "text": "Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum."
            }
        ],
        "created": moment.utc("2020-04-25").toDate()
    }
  ];

  //const ProductData: Array<ProductType> = []; // For no records test

  export default ProductData;