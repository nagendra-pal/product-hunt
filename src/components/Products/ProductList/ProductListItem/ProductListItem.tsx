import React, { useState } from "react";
import { Button, Modal, ModalHeader, ModalBody, ModalFooter } from "reactstrap";
import moment from "moment";
import {ProductListItemType} from "../../ProductTypes";
import {ProductContainer, ImageContainer, StyledImage, DetailsContainer, StyledProductName, StyledProductDate, ProductLikeContainer, StyledProductLike, StyledLikeCount} from "../../styles/Products";

const ProductListItem = ({ product }: ProductListItemType): JSX.Element => {
  const [likes, setLikes] = useState<number>(product.likes);
  const [modal, setModal] = useState<boolean>(false);
  const toggle = () => setModal(!modal);
  return (
    <div>
      <ProductContainer>
        <ImageContainer>
          <StyledImage onClick={toggle} src={product.image}></StyledImage>
        </ImageContainer>
        <DetailsContainer>
          <StyledProductName>{product.name}</StyledProductName>
          <div>{product.desc}</div>
          <StyledProductDate>
            {moment.utc(product.created).format("MMM DD, YYYY")}
          </StyledProductDate>
          <ProductLikeContainer>
            <StyledProductLike
              height="25"
              width="25"
              src="https://res.cloudinary.com/dsrkwxxo7/image/upload/v1589205364/samples/like-1_eu3iot.svg"
              onClick={() => setLikes(likes + 1)}
            ></StyledProductLike>
            <StyledLikeCount>{likes}</StyledLikeCount>
          </ProductLikeContainer>
        </DetailsContainer>
      </ProductContainer>
      <Modal isOpen={modal} toggle={toggle}>
        <ModalHeader toggle={toggle}>Comments</ModalHeader>
        <ModalBody>
          {product.comments.map((comment, index) => (
            <ul key={`comment-${index}`}>
              <li>{comment.text}</li>
            </ul>
          ))}
        </ModalBody>
        <ModalFooter>
          <Button color="secondary" onClick={toggle}>
            Cancel
          </Button>
        </ModalFooter>
      </Modal>
    </div>
  );
};

export default ProductListItem;
