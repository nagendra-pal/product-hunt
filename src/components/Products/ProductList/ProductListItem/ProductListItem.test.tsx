import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import ProductListItem from './ProductListItem';
import ProductData from "../../ProductData";

describe('Product List Item component', () => {
   it('should renders without crashing', () => {
      const output = shallow(<ProductListItem product={ProductData[0]} />);
      expect(shallowToJson(output)).toMatchSnapshot();
    });
});