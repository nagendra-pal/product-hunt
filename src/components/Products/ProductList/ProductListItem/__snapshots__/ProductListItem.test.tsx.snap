// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Product List Item component should renders without crashing 1`] = `
<div>
  <styled.div>
    <styled.div>
      <styled.img
        onClick={[Function]}
        src="https://res.cloudinary.com/dsrkwxxo7/image/upload/v1589218153/samples/products/Wireless-headphones_vxnlz2.jpg"
      />
    </styled.div>
    <styled.div>
      <styled.div>
        Wireless headphones
      </styled.div>
      <div>
        Lorem ipsum dolor amet post-ironic four dollar toasts selvage meh Lorem ipsum dolor amet post-ironic four dollar toasts selvage meh Lorem ipsum dolor amet post-ironic four dollar toasts selvage meh
      </div>
      <styled.div>
        Mar 25, 2020
      </styled.div>
      <styled.div>
        <styled.img
          height="25"
          onClick={[Function]}
          src="https://res.cloudinary.com/dsrkwxxo7/image/upload/v1589205364/samples/like-1_eu3iot.svg"
          width="25"
        />
        <styled.span>
          4
        </styled.span>
      </styled.div>
    </styled.div>
  </styled.div>
  <Modal
    autoFocus={true}
    backdrop={true}
    backdropTransition={
      Object {
        "mountOnEnter": true,
        "timeout": 150,
      }
    }
    centered={false}
    fade={true}
    isOpen={false}
    keyboard={true}
    modalTransition={
      Object {
        "timeout": 300,
      }
    }
    onClosed={[Function]}
    onOpened={[Function]}
    returnFocusAfterClose={true}
    role="dialog"
    scrollable={false}
    toggle={[Function]}
    unmountOnClose={true}
    zIndex={1050}
  >
    <ModalHeader
      charCode={215}
      closeAriaLabel="Close"
      tag="h5"
      toggle={[Function]}
      wrapTag="div"
    >
      Comments
    </ModalHeader>
    <ModalBody
      tag="div"
    >
      <ul
        key="comment-0"
      >
        <li>
          Lorem ipsum dolor sit amet, consectetur adipisicing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua.
        </li>
      </ul>
      <ul
        key="comment-1"
      >
        <li>
          Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat.
        </li>
      </ul>
      <ul
        key="comment-2"
      >
        <li>
          Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur.
        </li>
      </ul>
      <ul
        key="comment-3"
      >
        <li>
          Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.
        </li>
      </ul>
    </ModalBody>
    <ModalFooter
      tag="div"
    >
      <Button
        color="secondary"
        onClick={[Function]}
        tag="button"
      >
        Cancel
      </Button>
    </ModalFooter>
  </Modal>
</div>
`;
